### DISCUSSION TITLE
Archive YOURLS as a static website

### DISCUSSION THREAD

**Discussion by jota2rz** [YOURLS/YOURLS#3224] [Q&A]
I need to archive a YOURLS installation as a static website because no more shortened URLs will be created and it will be no longer maintained.
From a security and costs perspective this is the ideal approach.

Has anyone ever done this before?
The idea is to export each shortened URL as a index.html under it's own subfolder named as the short URL so it can be deployed as a static website.

**Comment by ozh** [comment]
That would be fairly easy : 

- export keyword => long urls from the DB (table `yourls_url`)
- make a script to generate each `folder/index.html` with the same HTML template doing the redirection
- remove all YOURLS files

From a SEO and browser perspective be aware that an HTML redirect is different than a proper redirect header, though (google "html redirect vs header" if you need more information)

**Reply by jota2rz** [reply]
I completely forgot about SEO, good catch.

And I have found a solution, since I'm currently using AWS S3 for everything static, there is a way to to make 301 redirects by just creating a zero-byte object and editing its metadata (https://docs.aws.amazon.com/AmazonS3/latest/userguide/how-to-page-redirect.html#redirect-requests-object-metadata).
